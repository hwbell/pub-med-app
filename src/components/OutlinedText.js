import React from 'react';
import '../App.scss';
import 'bootstrap/dist/css/bootstrap.css';

// components

// elements

// styling
// import ReactCSSTransitionGroup from 'react-addons-css-transition-group';

// elements

class OutlinedText extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {

    let alignSelf = this.props.alignLeft ? 'flex-start' : '';
    const styles = {
      content: {
        width: '100%',
        margin: '4px'
      },
      text: {
        paddingTop: '10px',
        alignSelf
      },
      title: {
        alignSelf
      }
    }

    return (
      <div className="outline" style={styles.content}>

        {this.props.title &&
          <p className="thread-title" style={styles.title}>{this.props.title}</p>
        }

        <p className="paragraph" style={styles.text}>{this.props.text}</p>

      </div>

    );
  }
}

export default OutlinedText;
