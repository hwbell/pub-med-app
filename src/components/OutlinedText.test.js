import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import OutlinedText from './OutlinedText';

describe('OutlinedText', () => {

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<OutlinedText />);
  })

  it('renders without crashing', async () => {
    wrapper.update();
  });
  
  it('renders correctly', async () => {
    const tree = await renderer
      .create(<OutlinedText />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it('contains the correct elements', () => {
    expect(wrapper.find('.outline').length).toBe(1);
    expect(wrapper.find('.paragraph').length).toBe(1);
  })

  it('should not show a title unless provided', () => {
    // without title prop there wont be a profile-title
    expect(wrapper.find('.profile-title').length).toBe(0);

    // once a title is provided it should show up
    wrapper.setProps({title: 'About'});
    wrapper.update();
    expect(wrapper.find('.profile-title').length).toBe(1);

  })

  it('should align items with alignLeft prop if provided', () => {
    // set a title so we can check this as well as the paragraph
    wrapper.setProps({title: 'About'});

    expect(wrapper.find('.profile-title').props().style.alignSelf).toBe('');
    expect(wrapper.find('.paragraph').props().style.alignSelf).toBe('');

    // set the prop
    wrapper.setProps({
      alignLeft: true
    });

    expect(wrapper.find('.profile-title').props().style.alignSelf).toBe('flex-start');
    expect(wrapper.find('.paragraph').props().style.alignSelf).toBe('flex-start');
  })
})
