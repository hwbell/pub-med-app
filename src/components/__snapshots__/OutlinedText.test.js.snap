// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutlinedText renders correctly 1`] = `
<div
  className="outline"
  style={
    Object {
      "margin": "4px",
      "width": "100%",
    }
  }
>
  <p
    className="paragraph"
    style={
      Object {
        "alignSelf": "",
        "paddingTop": "10px",
      }
    }
  />
</div>
`;
